docs/api.rst

```diff
@@ -189,13 +189,70 @@ It allows you to maintain and update a "state" (aka feedback, aka seed for the n
   	return { seed: values, value: avg }
   }, [], stream)
 
-zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+.. _zipArrayValues 
 
-withArrayValues :: [a] -> Stream b -> Stream a
+zipArrayValues
+^^^^^^^^^^^^^^
 
-chain :: (a -> Stream b) -> Stream a -> Stream b
+.. code-block:: haskell
+
+  zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+
+Creates a new stream by applying a function ``((a, b) -> c)`` to the latest value of ``Stream b`` with an increasing index of ``[a]`` until ``Stream b`` ends or there are no more values in ``[a]``.::
```

Comment:
I agree, I find the `creates a new stream ...` to be verbose as well. I was just taking lead from some of the other documentation. 

I also agree with your concerns about "ends when ..." too 👍